src/library.js

```diff
@@ -448,6 +448,9 @@ LibraryManager.library = {
   // TODO: There are currently two abort() functions that get imported to asm module scope: the built-in runtime function abort(),
   // and this function _abort(). Remove one of these, importing two functions for the same purpose is wasteful.
   abort__sig: 'v',
+#if !MINIMAL_RUNTIME
+  abort__proxy: 'sync',
```

Comment:
```suggestion
  // Proxy synchronously, which will also have the effect of halting the pthread.
  // (Async proxying would let the pthread continue until the main thread does
  // something.)
  abort__proxy: 'sync',
```

Reply:
I'm not sure why that comment would be needed here by not anywhere else where we use `abort__proxy`.   Its seem clear why we would want proxy this and all syscalls as `"sync"`.  In fact there is no current usage of `"async"` in the library code.   Given that, I'm not sure why we even have that as an option for library code

Reply:
Interesting that we have no async proxying in `library*.js`, I thought we did... maybe it was removed.

While we do still support async proxying, though, I think the comment is important. I spent some time here puzzling out why this PR was valid, as it seemed to me like it would halt the main thread but not this thread. Then I realized the sync proxying replaces an abort on this thread with a hang. That seems good enough, but I think clarifying that is good, at least for me...

Reply:
The other reason this works is that, when abort it called on the main thread it will terminate all the worker threads (including this one).   At least for normal runtime (I think MINIMAL_RUNTIME has a bug where PThread.terminateAllThreads is missing from exit/abort).

Reply:
> I think MINIMAL_RUNTIME has a bug where PThread.terminateAllThreads is missing from exit/abort

I think PR #14481 will resolve that (it ensures that the ` PThread.terminateAllThreads()` call is also added in `postamble_minimal.js`).

Reply:
@sbc100 True, but terminating the worker threads is not immediate, both because of the time it takes to proxy to the main thread (which is fire-and-forget if async), and also, there is no guarantee on when the browser actually terminates threads when requested to. So async proxying would give this thread time to continue forward for an unspecified time, which could be really weird.

Reply:
Yes clearly async proxying would make no sense at all here.   (The same is true for all syscall though.. they are expected to by synchronous and would be completely broken in async mode).

Reply:
True. Outside of syscalls though we do use async proxying for several things, like WebGL. It's just by chance that none of that is in `library*.js` files.